firecracker-control/local.go

```diff
@@ -348,3 +402,36 @@ func (s *local) getVM(id string) (*instance, error) {
 
 	return instance, nil
 }
+
+// PatchStubDrive will replace a stub drive in the queue with a new path.
+func (s *local) PatchStubDrive(ctx context.Context, req *proto.PatchStubDriveRequest) (*proto.PatchStubDriveResponse, error) {
+	if s.stubDriveIndex >= len(s.drives) {
+		return nil, ErrStubDrivesExhausted
+	}
+
+	d := s.drives[s.stubDriveIndex]
+	d.PathOnHost = &req.PathOnHost
+
+	if d.DriveID == nil {
+		// this should never happen, but we want to ensure that we never nil
+		// dereference
+		return nil, ErrDriveIDNil
+	}
+
+	s.drives[s.stubDriveIndex] = d
+
+	instance, err := s.getVM(req.GetVMID())
+	if err != nil {
+		return nil, err
+	}
+
+	err = instance.machine.UpdateGuestDrive(ctx, firecracker.StringValue(d.DriveID), req.PathOnHost)
+	if err != nil {
+		return nil, err
+	}
+
+	s.stubDriveIndex++
```

Comment:
Can we move fake drive management logic to a separate drive manager/handler object? I think we don't want to leak implementation details and it'll be easier to manage locks, and easier to replace implementation when FC will improve drive handling.